andela/codepirates-ah-backend
# #167313402  user profile CRUD operation

#### What does this PR do?

- create  user profile 

#### Description of Task to be completed?

- user should be able to create a profile

- user should be able to update their profile

- user should be able to view other peoples profiles 

#### How should this be manually tested?

- by cloning this repo and switch to this branch 

- in your dotenv file, you should have cloudinary configuration including  `CLOUDINARY_NAME`,
  `CLOUDINARY_API_KEY`,
 `CLOUDINARY_SECRET_KEY`,

- run migration and seed then dev scripts 

- after you can login and use token to create a profile by making  a PUT request to  `/profile`  route

- you can view your profile by making  a GET request to `/profile/ `route 

- you can view someone profile by making  a GET request to `/profile/: username`   route 

#### Any background context you want to provide?

#### What are the relevant pivotal tracker stories?
[#167313402](https://www.pivotaltracker.com/story/show/167313402)

#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/38810299/63132555-bf43dd80-bfc1-11e9-8557-0d6209a75570.png)
![image](https://user-images.githubusercontent.com/38810299/63132561-c7038200-bfc1-11e9-854f-28c0e59ff990.png)
![image](https://user-images.githubusercontent.com/38810299/63132569-cd91f980-bfc1-11e9-8899-7b98fe90d292.png)
![image](https://user-images.githubusercontent.com/38810299/63132574-d4b90780-bfc1-11e9-8ced-6f8b1914038b.png)
![image](https://user-images.githubusercontent.com/38810299/63132628-1a75d000-bfc2-11e9-928e-2cc4dc402399.png)



#### Questions:
